server/annotations.go

```diff
@@ -266,6 +305,13 @@ func (s *Service) NewAnnotation(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if req.Tags != nil {
```

Comment:
I think we might not need to check if `req.Tags != nil` here and we can just call `Valid()`.